joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/passes/ImplicitRequirePass.scala

```diff
@@ -40,37 +40,48 @@ class ImplicitRequirePass(cpg: Cpg, programSummary: RubyProgramSummary) extends
   /** Collects methods within a module.
     */
   private def findMethodsViaAstChildren(module: Method): Iterator[Method] = {
-    Iterator(module) ++ module.astChildren.flatMap {
-      case x: TypeDecl => x.method.flatMap(findMethodsViaAstChildren)
-      case x: Method   => Iterator(x) ++ x.astChildren.collectAll[Method].flatMap(findMethodsViaAstChildren)
-      case _           => Iterator.empty
-    }
+    // TODO For now we have to go via the full name regex because the AST is not yet linked
+    // at the execution time of this pass.
+    // Iterator(module) ++ module.astChildren.flatMap {
+    //  case x: TypeDecl => x.method.flatMap(findMethodsViaAstChildren)
+    //  case x: Method   => Iterator(x) ++ x.astChildren.collectAll[Method].flatMap(findMethodsViaAstChildren)
+    //  case _           => Iterator.empty
+    // }
+    cpg.method.fullName(module.fullName + ".*")
   }
 
   override def runOnPart(builder: DiffGraphBuilder, part: Method): Unit = {
-    findMethodsViaAstChildren(part).ast.isCall
-      .flatMap {
-        case x if x.name == Operators.alloc =>
-          x.argument.isIdentifier
-        case x =>
-          x.receiver.fieldAccess.fieldIdentifier
-      }
-      .map {
-        case fi: FieldIdentifier => fi -> programSummary.matchingTypes(fi.canonicalName)
-        case i: Identifier       => i  -> programSummary.matchingTypes(i.name)
-      }
-      .distinct
-      .foreach { case (identifier, rubyTypes) =>
+    val identifiersToMatch = mutable.ArrayBuffer.empty[String]
+
+    val typeDecl = cpg.typeDecl.fullName(part.fullName + ".*").l
+    typeDecl.inheritsFromTypeFullName.foreach(identifiersToMatch.append)
+
+    val methods = findMethodsViaAstChildren(part).toList
+    val calls   = methods.ast.isCall.toList
+    val identifiers = calls.flatMap {
+      case x if x.name == Operators.alloc =>
+        x.argument.isIdentifier.name
+      case x =>
+        x.receiver.fieldAccess.fieldIdentifier.canonicalName
```

Comment:
`.fieldAccess` likely doesn't do what we want here. It returns any field access recurively found below the `receiver`.

We either want to do `.isCall.nameExact(Operators.fieldAccess)`, or finally add a new helper that isn't just meant for the REPL.

Reply:
Wtf..., thanks for catching this. This kind of accessors should be completely deleted...

Reply:
Done

Reply:
I wouldn't mind removing them. I think there are some (potentially buggy) uses in the joern code though, that'd need to be checked and adapted.